src/components/network-map-tab.js

```diff
@@ -209,16 +221,127 @@ export const NetworkMapTab = ({
         []
     );
 
-    const loadMapGeoData = useCallback(() => {
+    const getEquipmentsNotFoundIds = useCallback(
+        (foundEquipmentPositions, allEquipments) => {
+            const foundEquipmentsIds = Array.from(
+                foundEquipmentPositions.keys()
+            );
+
+            const notFoundEquipmentsIds = allEquipments
+                .filter(
+                    (s) =>
+                        !foundEquipmentsIds.includes(s.id) ||
+                        temporaryGeoData.temporaryGeoDataIds.has(s.id)
+                )
+                .map((s) => s.id);
+
+            return notFoundEquipmentsIds;
+        },
+        [temporaryGeoData.temporaryGeoDataIds]
+    );
+
+    const getMissingEquipmentsPositions = useCallback(
+        (notFoundEquipmentsIds, fetchEquipmentCB) => {
+            if (notFoundEquipmentsIds.length === 0) {
+                return Promise.resolve([]);
+            }
+
+            return fetchEquipmentCB(
+                studyUuid,
+                currentNode?.id,
+                notFoundEquipmentsIds
+            );
+        },
+        [studyUuid, currentNode?.id]
+    );
+
+    const loadMissingGeoData = useCallback(() => {
+        const notFoundSubstationIds = getEquipmentsNotFoundIds(
+            geoData.substationPositionsById,
+            mapEquipments.getSubstations()
+        );
+        temporaryGeoData.addGeoDataIds(notFoundSubstationIds);
+
+        const notFoundLinesIds = lineFullPath
+            ? getEquipmentsNotFoundIds(
+                  geoData.linePositionsById,
+                  mapEquipments.getLines()
+              )
+            : [];
+        temporaryGeoData.addGeoDataIds(notFoundLinesIds);
+
+        if (notFoundSubstationIds.length > 0 || notFoundLinesIds.length > 0) {
+            console.info(
+                `Loading geo data of study '${studyUuid}' of missing substations '${notFoundSubstationIds}' and missing lines '${notFoundLinesIds}'...`
+            );
+
+            setWaitingTemporaryLoadGeoData(true);
+
+            const missingSubstationPositions = getMissingEquipmentsPositions(
+                notFoundSubstationIds,
+                fetchSubstationPositions
+            );
+
+            const missingLinesPositions = getMissingEquipmentsPositions(
+                notFoundLinesIds,
+                fetchLinePositions
+            );
+
+            Promise.all([missingSubstationPositions, missingLinesPositions])
+                .then((positions) => {
+                    if (positions[0].length > 0 || positions[1].length > 0) {
+                        geoData.addSubstationPositions(positions[0]);
+                        geoData.addLinePositions(positions[1]);
+                        // If there is new substation positions, we instantiate a new Map so that the substations layer rendering is triggered.
+                        // Same for line positions.
+                        const newGeoData = new GeoData(
```

Comment:
Add "deep" compare to avoid recreating the maps and the geodata when nothing has changed (the data in the http request matches what we already had)

Reply:
Done